vcr/issuer/interface.go

```diff
@@ -0,0 +1,43 @@
+package issuer
+
+import (
+	ssi "github.com/nuts-foundation/go-did"
+	"github.com/nuts-foundation/go-did/did"
+	"github.com/nuts-foundation/go-did/vc"
+	"github.com/nuts-foundation/nuts-node/crypto"
+	"github.com/nuts-foundation/nuts-node/vcr/credential"
+)
+
+// Publisher publishes new credentials and revocations to a channel. Used by a credential issuer.
+type Publisher interface {
+	PublishCredential(verifiableCredential vc.VerifiableCredential, public bool) error
+	PublishRevocation(revocation credential.Revocation) error
+}
+
+type keyResolver interface {
```

Comment:
why not use the VDR `KeyResolver`?

Reply:
VDR only has access to public keys embedded in the docs. This interface defines a service which can resolve private keys for a DID.